.github/workflows/publish_pypi.yml

```diff
@@ -2,251 +2,36 @@ name: Build and publish package to PyPI
-              for file in {dest_dir}/*.whl; do mv "$file" "${file//macosx_11_1/macosx_11_0}"; done
-            fi
-          CIBW_TEST_EXTRAS: "all,dev,jax"
-          CIBW_TEST_COMMAND: |
-            set -e -x
-            python -m pytest -m cibw {project}/tests/unit
-
-      - name: Upload wheels for macOS (amd64, arm64)
-        uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
-        with:
-          name: wheels_${{ matrix.os }}
-          path: ./wheelhouse/*.whl
-          if-no-files-found: error
-
-  build_sdist:
-    name: Build SDist
-    runs-on: ubuntu-latest
-
-    steps:
-      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
-      - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
-        with:
-          python-version: 3.12
-
-      - name: Build SDist
-        run: pipx run build --sdist
-
-      - name: Upload SDist
-        uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
-        with:
-          name: sdist
-          path: ./dist/*.tar.gz
-          if-no-files-found: error
-
-  publish_pypi:
-    # This job is only of value to PyBaMM and would always be skipped in forks
-    if: github.event_name != 'schedule' && github.repository == 'pybamm-team/PyBaMM'
-    name: Upload package to PyPI
-    needs: [
-      build_manylinux_wheels,
-      build_macos_wheels,
-      build_windows_wheels,
-      build_sdist
-    ]
-    runs-on: ubuntu-latest
-    environment:
-      name: pypi
-      url: https://pypi.org/p/pybamm
+        python-version: ["3.12"]
     permissions:
       id-token: write
 
     steps:
-      - name: Download all artifacts
-        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
         with:
-          path: artifacts
-          merge-multiple: true
+          python-version: ${{ matrix.python-version }}
 
-      - name: Sanity check downloaded artifacts
-        run: ls -lA artifacts/
+      - name: Build wheel
+        run: |
+          pip wheel . --no-deps -w deploy
 
-      - name: Publish to PyPI
-        if: github.event.inputs.target == 'pypi' || github.event_name == 'release'
-        uses: pypa/gh-action-pypi-publish@67339c736fd9354cd4f8cb0b744f2b82a74b5c70 # v1.12.3
+      - name: Upload package
+        uses: actions/upload-artifact@v4
         with:
-          packages-dir: artifacts/
+          name: wheels
+          path: deploy/
+          if-no-files-found: error
 
-      - name: Publish to TestPyPI
-        if: github.event.inputs.target == 'testpypi'
-        uses: pypa/gh-action-pypi-publish@67339c736fd9354cd4f8cb0b744f2b82a74b5c70 # v1.12.3
+      - name: Publish on PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
It is strongly recommended to split the publishing action to a new job that `needs:` the build job for security reasons, and add the `id-token:` permission to just the publishing job: https://github.com/pypa/gh-action-pypi-publish#trusted-publishing and https://docs.pypi.org/trusted-publishers/security-model/#provider-specific-considerations. Also, we'll need an `environment:` for the publishing job.